File: agent/macos/Sources/FenrirAgent/AgentConfig.swift
import Foundation

struct AgentConfig {
    let backendURL: URL
    let deviceId: String
    let logLevel: String

    static func load() -> AgentConfig {
        let deviceId = Host.current().localizedName ?? "Unknown-Mac"

        let backendURLString = ProcessInfo.processInfo.environment["FENRIR_BACKEND_URL"]
            ?? "http://localhost:3000/api/v1/telemetry"
        let backendURL = URL(string: backendURLString)!

        return AgentConfig(
            backendURL: backendURL,
            deviceId: deviceId,
            logLevel: ProcessInfo.processInfo.environment["FENRIR_LOG_LEVEL"] ?? "INFO"
        )
    }
}

File: agent/linux/pkg/config/AgentConfig.go
package config

import (
	"os"
)

type AgentConfig struct {
	BackendURL string
	DeviceId   string
	LogLevel   string
}

func LoadConfig() *AgentConfig {
	deviceId, err := os.Hostname()
	if err != nil {
		deviceId = "Unknown-Linux"
	}

	backendURL := os.Getenv("FENRIR_BACKEND_URL")
	if backendURL == "" {
		backendURL = "http://localhost:3000/api/v1/telemetry"
	}

	return &AgentConfig{
		BackendURL: backendURL,
		DeviceId:   deviceId,
		LogLevel:   "INFO",
	}
}
